pallets/cash/src/internal/validate_trx.rs

```diff
@@ -0,0 +1,570 @@
+use crate::{
+    chains::{Chain, ChainAccount, Ethereum},
+    internal,
+    notices::EncodeNotice,
+    AllowedNextCodeHash, Call, Config, Notices, Validators,
+};
+use codec::Encode;
+use frame_support::storage::{IterableStorageMap, StorageDoubleMap, StorageValue};
+use our_std::RuntimeDebug;
+use sp_runtime::transaction_validity::{TransactionSource, TransactionValidity, ValidTransaction};
+
+#[derive(Eq, PartialEq, RuntimeDebug)]
+pub enum ValidationError {
+    InvalidNextCode,
+    InvalidValidator,
+    InvalidSignature,
+    InvalidCall,
+    InvalidPriceSignature,
+    UnknownNotice,
+}
+
+pub fn validate_unsigned<T: Config>(
+    _source: TransactionSource,
+    call: &Call<T>,
+) -> Result<TransactionValidity, ValidationError> {
+    match call {
+        Call::set_miner(_miner) => Ok(ValidTransaction::with_tag_prefix("Gateway::set_miner")
+            .priority(1)
+            .longevity(10)
```

Comment:
surprisign to see set_miner here, isn't it set in inherent? Longevity also seems weird, as if the tx's could float around in the pool and maybe it doesn't get mined the next bvlock and someone gets 2 rewards in a row for authoring one block?

Reply:
I think we should actually check `TransactionSource` is internal, maybe with a few others, as well.

Reply:
I updated with better usage of transaction source here.